projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMSignal.java

```diff
@@ -0,0 +1,245 @@
+            return LLVM_SIG_ERR;
+        }
+
+        return returnFunction;
+    }
+
+    // TODO: stack handling should work without predefined sizes,...
+    private static final long TMP_SIGNAL_STACK_SIZE_BYTE = 1 * 1024;
+
+    private static final class LLVMSignalHandler implements SignalHandler {
+
+        private final LLVMFunctionDescriptor function;
+        private RootCallTarget callTarget;
+
+        private LLVMSignalHandler(Signal signal, LLVMFunctionDescriptor function) throws IllegalArgumentException {
+            this.function = function;
+            LLVMFunctionNode functionNode = LLVMFunctionLiteralNodeGen.create(function);
+
+            // TODO: currently we create a new stack with a specific size, this has to be changed
+            LLVMAddressLiteralNode signalStack = new LLVMAddressLiteralNode(LLVMAddress.fromLong(LLVMStack.allocate(TMP_SIGNAL_STACK_SIZE_BYTE)));
+            LLVMI32LiteralNode sigNumArg = new LLVMI32LiteralNode(signal.getNumber());
+            LLVMExpressionNode[] args = {signalStack, sigNumArg};
+
+            LLVMContext context = LLVMLanguage.INSTANCE.findContext0(LLVMLanguage.INSTANCE.createFindContextNode0());
+
+            LLVMUnresolvedCallNode callNode = new LLVMUnresolvedCallNode(functionNode, args, LLVMRuntimeType.VOID, context);
+
+            callTarget = Truffle.getRuntime().createCallTarget(
+                            new LLVMFunctionStartNode(callNode,
+                                            new LLVMNode[]{},
+                                            new LLVMNode[]{},
+                                            SourceSection.createUnavailable("", null),
+                                            new FrameDescriptor(), ""));
+
+            if (function.equals(LLVM_SIG_DFL)) {
+                Signal.handle(signal, SignalHandler.SIG_DFL);
+            } else if (function.equals(LLVM_SIG_IGN)) {
+                Signal.handle(signal, SignalHandler.SIG_IGN);
+            } else {
+                Signal.handle(signal, this);
+            }
+        }
+
+        @Override
+        public void handle(Signal arg0) {
+            /*
+             * https://en.wikipedia.org/wiki/Sigaction#Replacement_of_deprecated_signal.28.29
+             *
+             * Signal handlers installed by the signal() interface will be uninstalled immediately
+             * prior to execution of the handler.
+             *
+             * @note I made such a program, and I had to delete the handler manually, so this
+             * sentence is probably not true
+             */
+            callTarget.call();
+        }
+
+        public LLVMFunctionDescriptor getFunction() {
+            return function;
+        }
+
+    }
+
+    /**
+     * Handling conversation from Signal number to Signal Object in a clean way.
+     *
+     * documentation: http://man7.org/linux/man-pages/man7/signal.7.html
+     */
+    private enum Signals {
+
+        // POSIX.1-1990
+        SIG_HUP("HUP"),
```

Comment:
I don't rely on the sequence of the signals. To be more specific, only a few signals even have a specified signal number defined in the POSIX standard (https://en.wikipedia.org/wiki/Unix_signal#POSIX_signals)

Because of that, the signal number, as well as which signals are supported on the system is evaluated at runtime in this enum.